pkg/provisioner/ironic/ironic.go

```diff
@@ -1399,6 +1418,13 @@ func (p *ironicProvisioner) hardPowerOff() (result provisioner.Result, err error
 			result.Dirty = true
 			return result, nil
 		}
+		// If the target state is unset while the last error is set,
+		// then the last execution of hard power off has failed.
```

Comment:
Won't this also be the case if the last target state was soft power off and execution of that failed? If so then this will break the fallback from soft power off to hard power off in regular operation.

Reply:
Agreed. This is probably unnecessary.